apps/web/pages/[user].tsx

```diff
@@ -225,20 +239,31 @@ export type UserPageProps = {
 
 export const getServerSideProps: GetServerSideProps<UserPageProps> = async (context) => {
   const ssr = await ssrInit(context);
-  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(context.req.headers.host ?? "");
-
+  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(
+    context.req.headers.host ?? "",
+    context.query.orgSlug
+  );
   const usernameList = getUsernameList(context.query.user as string);
   const dataFetchStart = Date.now();
   const usersWithoutAvatar = await prisma.user.findMany({
     where: {
       username: {
         in: usernameList,
       },
-      organization: isValidOrgDomain
-        ? {
-            slug: currentOrgDomain,
-          }
-        : null,
+      organization:
+        isValidOrgDomain && currentOrgDomain
+          ? {
+              OR: [
+                { slug: currentOrgDomain },
+                {
+                  metadata: {
+                    path: ["requestedSlug"],
+                    equals: currentOrgDomain,
```

Comment:
Needed to make sure we also selected an unpublished organization to not end up in a 404 page